versioned/persist/tx/src/main/java/org/projectnessie/versioned/persist/tx/TxDatabaseAdapter.java

```diff
@@ -175,18 +176,7 @@ public Map<Key, ContentAndState<ByteString>> values(
 
   @Override
   public Stream<CommitLogEntry> commitLog(Hash offset) throws ReferenceNotFoundException {
-    ConnectionWrapper conn = borrowConnection();
-    boolean failed = true;
-    try {
-      Stream<CommitLogEntry> intLog = readCommitLogStream(conn, offset);
-
-      failed = false;
-      return intLog.onClose(conn::close);
-    } finally {
-      if (failed) {
-        conn.close();
-      }
-    }
+    return withConnectionWrapper(conn -> readCommitLogStream(conn, offset));
```

Comment:
`withConnectionWrapper` is a nice solution to reduce the redundant code!

however it feels like `commitLog` and the other methods should still be annotated with `@MustBeClosed` in this file?

or is this correctly "inherited" through the `AbstractDatabaseAdapter` base class?
if yes, this seems to disagree with what we are doing in `NonTransactionalDatabaseAdapter`

Reply:
Not sure how errorprone exactly determines that. The detection feels a bit "broken" - but OTOH it's due to how annotations are inherited (they're not) So it seems taht errorprone requies an explicit `@MustBeClosed` in some situations, even if the declaring base method already has a  `@MustBeClosed`.

Another actual bug is that errorrpone requires an `@SuppressWarnings("MustBeClosedChecker")` for e.g. `Stream.flatMap()` - returning a closed the stream from `flatMap()` is just ... wrong.